src/pages/game.tsx

```diff
@@ -0,0 +1,84 @@
+import { useEffect, useRef } from 'react';
+import Field from '../api/Field';
+import shuffle from '../api/shuffleArray';
+import filedSize15 from '../api/dimensions';
+import Content from '../components/Content';
+
+export default function GamePage() {
+  const sideSize: number = filedSize15.sideSize as number;
+
+  const numbers: number[] = shuffle(filedSize15.arr as number[]);
+
+  const ref = useRef<HTMLCanvasElement>(null);
+
+  useEffect(() => {
+    const ctx: CanvasRenderingContext2D = ref.current?.getContext('2d') as CanvasRenderingContext2D;
+    if (!ctx) throw new Error('Error');
+
+    const rect: DOMRect = ref.current?.getBoundingClientRect() as DOMRect;
+
+    const field = new Field(sideSize, 'lightgreen', ctx, [], numbers);
+    field.spawnTiles();
+
+    const handleClick = (e: MouseEvent) => {
+      const x = e.clientX - rect.left;
+      const y = e.clientY - rect.top;
+      const idx = field.getTileIndex(x, y);
+
+      if (field.tiles[idx - 1]) {
+        if (field.tiles[idx - 1].num === 0 && idx !== 0 && idx !== 4 && idx !== 8 && idx !== 12) {
+          field.updateTile(idx, 'left');
+          return;
+        }
+      }
+
+      if (field.tiles[idx + 1]) {
+        if (field.tiles[idx + 1].num === 0 && idx !== 3 && idx !== 7 && idx !== 11 && idx !== 15) {
+          field.updateTile(idx, 'right');
+          return;
+        }
+      }
+
+      if (field.tiles[idx - 4]) {
+        if (field.tiles[idx - 4].num === 0 && idx !== 0 && idx !== 1 && idx !== 2 && idx !== 3) {
+          field.updateTile(idx, 'top');
+          return;
+        }
+      }
+
+      if (field.tiles[idx + 4]) {
+        // eslint-disable-next-line max-len
+        if (field.tiles[idx + 4].num === 0 && idx !== 12 && idx !== 13 && idx !== 14 && idx !== 15) {
```

Comment:
Лучше переносить код, чем отключать max-len правило, часто потом при ревью код вылазит за пределы фрейма и появляется горизонтальный скроллинг, плюс читаемость уменьшается значительно. Все равно при билде вебпак все переносы грохнет, так что нет смысла экономить строки.
```suggestion
        if (field.tiles[idx + 4].num === 0
          && idx !== 12
          && idx !== 13
          && idx !== 14
          && idx !== 15
        ) {
```

А еще если idx не может быть больше 15, то, возможно, будет достаточно проверить idx < 12, но я пока в математику тут сильно не вникал, поэтому это не точно